src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -380,21 +376,22 @@ public static string GetTypeInfoName(ITypeInfo typeInfo)
         /// ReleaseComObject on a RCW and not worry about other active uses ofsaid RCW.
         /// </summary>
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern object GetUniqueObjectForIUnknown(IntPtr unknown);
+        public static extern object? GetUniqueObjectForIUnknown(IntPtr unknown);
```

Comment:
This does not look right. When this can return null?

Reply:
I called this out here: https://github.com/dotnet/coreclr/pull/23699#discussion_r271908202

It looks like `MarshalNative::GetUniqueObjectForIUnknown` just forwards to `GetObjectRefFromComIP`. `GetObjectRefFromComIP` then doesn't look to ensure that `pObjOut` is non-null for every potential code path; but I was unsure if that can actually happen in practice and it looked like for the `GetUniqueObjectForIUnknown` case, it should always try to find or create an object ref if one doesn't exist yet...

Reply:
Right. If you pass non-null punk to `GetObjectRefFromComIP. GetObjectRefFromComIP`, it will give you non-null object back.

Reply:
Fixed.